File: android/app/src/main/java/com/ayuvo/health/models/FoodProductMetadata.kt
package com.ayuvo.health.models

import kotlinx.serialization.Serializable

/** Optional product-label details returned by Open Food Facts for barcode logs. */
@Serializable
data class FoodProductMetadata(
    val barcode: String,
    val packageQuantity: String? = null,
    val ingredientsText: String? = null,
    val allergens: List<String> = emptyList(),
    val traces: List<String> = emptyList(),
    val nutriScore: String? = null,
    val novaGroup: Int? = null,
    val ecoScore: String? = null,
    val labels: List<String> = emptyList(),
    val categories: List<String> = emptyList(),
    val imageUrl: String? = null
) {
    val hasDisplayDetails: Boolean
        get() = barcode.isNotEmpty() ||
            packageQuantity != null ||
            ingredientsText != null ||
            allergens.isNotEmpty() ||
            traces.isNotEmpty() ||
            nutriScore != null ||
            novaGroup != null ||
            ecoScore != null ||
            labels.isNotEmpty() ||
            categories.isNotEmpty()
}

File: ios/calorietracker/Models/FoodProductMetadata.swift
import Foundation

/// Product-label details supplied by Open Food Facts for a barcode lookup.
/// Nutrition remains on `FoodEntry`; this type keeps optional product metadata
/// separate so legacy and non-barcode food logs stay lightweight.
struct FoodProductMetadata: Codable, Equatable, Sendable {
    var barcode: String
    var packageQuantity: String?
    var ingredientsText: String?
    var allergens: [String]
    var traces: [String]
    var nutriScore: String?
    var novaGroup: Int?
    var ecoScore: String?
    var labels: [String]
    var categories: [String]
    var imageURL: URL?

    var hasDisplayDetails: Bool {
        !barcode.isEmpty
            || packageQuantity != nil
            || ingredientsText != nil
            || !allergens.isEmpty
            || !traces.isEmpty
            || nutriScore != nil
            || novaGroup != nil
            || ecoScore != nil
            || !labels.isEmpty
            || !categories.isEmpty
    }
}
